src/dotnet/Db/Module/DbModule.cs

```diff
@@ -58,8 +61,61 @@ public class DbModule : HostModule<DbSettings>
         };
 
         // Adding services
+        services.TryAddSingleton<OutstandingDbConnectionsCounter>();
         services.AddSingleton(dbInfo);
-        services.AddDbContextFactory<TDbContext>(builder => {
+        services.AddDbContextFactory<TDbContext>((svp, builder) => {
+            var counter = svp.GetRequiredService<OutstandingDbConnectionsCounter>();
```

Comment:
I would actually move this code to some dedicated type.

Reply:
I will remove all this code from DbModule after I complete connections leak investigation.